hazelcast/src/main/java/com/hazelcast/config/NearCachePreloaderConfig.java

```diff
@@ -0,0 +1,197 @@
+/*
+ * Copyright (c) 2008-2016, Hazelcast, Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.config;
+
+import com.hazelcast.nio.ObjectDataInput;
+import com.hazelcast.nio.ObjectDataOutput;
+import com.hazelcast.nio.serialization.DataSerializable;
+import com.hazelcast.nio.serialization.impl.BinaryInterface;
+import com.hazelcast.spi.annotation.Beta;
+import com.hazelcast.spi.annotation.PrivateApi;
+
+import java.io.IOException;
+import java.io.Serializable;
+
+import static com.hazelcast.util.Preconditions.checkNotNull;
+import static com.hazelcast.util.Preconditions.checkPositive;
+
+/**
+ * Configuration for eviction.
+ * You can set a limit for number of entries or total memory cost of entries.
+ */
+@Beta
+@BinaryInterface
+public class NearCachePreloaderConfig implements DataSerializable, Serializable {
+
+    /**
+     * Default initial delay for the Near Cache key storage.
+     */
+    public static final int DEFAULT_STORE_INITIAL_DELAY_SECONDS = 600;
+
+    /**
+     * Default interval for the Near Cache key storage (in seconds).
+     */
+    public static final int DEFAULT_STORE_INTERVAL_SECONDS = 600;
+
+    private boolean enabled;
+    private String fileName = "";
+    private int storeInitialDelaySeconds = DEFAULT_STORE_INITIAL_DELAY_SECONDS;
+    private int storeIntervalSeconds = DEFAULT_STORE_INTERVAL_SECONDS;
+
+    private NearCachePreloaderConfig readOnly;
+
+    public NearCachePreloaderConfig() {
+    }
+
+    public NearCachePreloaderConfig(NearCachePreloaderConfig nearCachePreloaderConfig) {
+        /**
+         * ===== NOTE =====
```

Comment:
I copied that style from all other config classes. I don't like it because it's a dangling JavaDoc comment, which is actually not allowed at that position. But I would prefer to have all config classes in the same style. So I would merge it like this an clean up all config classes at once, if we can agree on a better style.